Kernel/Locking/LockRank.h

```diff
@@ -34,6 +34,9 @@ enum class LockRank : int {
     // Process locks are the highest rank, as they normally are taken
     // first thing when processing syscalls.
     Process = 0x010,
+
+    // Mutexes need to be taken before spinlocks, so they get their own lock-rank
+    Mutex = 0x020,
```

Comment:
Note to self: We should reorganize this bitfield such that lowest 16 bits are reserved for spinlocks (and 0x8000 is the Process lock) and highest 16 bits are reserved for mutexes. If you want you can already make this 0x1'0000 even though the circular dependency prevents any rank differentiation on Mutex right now.

Reply:
Currently Mutexes don't have any lasting effect on the lock rank, as this technically is the lock rank of the internal spinlock of the mutex...